scrapper.py

```diff
@@ -31,67 +42,176 @@ class Crawler:
+
+            urls_number = min(max_articles_num_per_seed, len(page_urls), (max_articles_num - len(self.urls)))
+            for index in range(urls_number):
+                self.urls.append('https://vn.ru' + self._extract_url(article_bs=page_urls[index]))
 
     def get_search_urls(self):
         """
         Returns seed_urls param
         """
-        pass
+
+        themes = ['ekonomika', 'oblast', 'obshchestvo',  'proisshestviya', 'dom', 'transport', 'razvlecheniya']
+        self.seed_urls.extend([f'{self.seed_urls[0]}news/{theme}/' for theme in themes])
 
 
 class ArticleParser:
     """
     ArticleParser implementation
     """
+
     def __init__(self, full_url: str, article_id: int):
-        pass
+        self.full_url = full_url
+        self.article_id = article_id
+        self.article = Article(full_url, article_id)
 
     def _fill_article_with_text(self, article_soup):
-        pass
+
+        article_text = article_soup.find('div', {'class': 'js-mediator-article divider-news clearfix'}).find_all('p')
+
+        if not re.sub(r'[^А-Яа-я]', '', ''.join(list(map(lambda x: x.text, article_text)))):
+            text = article_soup.find('div', {'class': 'js-mediator-article divider-news clearfix'}).contents
+            text_list = []
+            for item in text:
+                if isinstance(item, NavigableString):
+                    continue
+
+                for index in range(len(item.contents)):
+                    if isinstance(item.contents[index], NavigableString):
+                        text_list.append(item.contents[index])
+        else:
+            text_list = list(map(lambda x: x.text, article_text))
+
+        self.article.text = '\n'.join(text_list)
 
     def _fill_article_with_meta_information(self, article_soup):
-        pass
+        self.article.title = article_soup.find('h1').text
+
+        self.article.author = article_soup.find('div', {'class': 'author-name floatleft clearfix'}).text.strip('\n')
+
+        raw_topics = article_soup.find_all('div', {'class': 'on-footer-row'})
+        self.article.topics = list(map(lambda x: x.find('a').text.lower(), raw_topics))
+
+        raw_date = article_soup.find('div', {'class': 'nw-dn-date floatleft'}).text
+        self.article.date = self.unify_date_format(raw_date)
 
     @staticmethod
     def unify_date_format(date_str):
         """
         Unifies date format
         """
-        pass
+
+        return datetime.datetime.strptime(date_str, '%d.%m.%Y')
 
     def parse(self):
         """
         Parses each article
         """
-        pass
+
+        response = requests.get(self.full_url, headers=headers)
+
+        if not response:
+            raise IncorrectURLError
+
+        page_content = response.content
+        article_bs = BeautifulSoup(page_content, features='lxml')
+
+        self._fill_article_with_text(article_bs)
+        self._fill_article_with_meta_information(article_bs)
+        self.article.save_raw()
```

Comment:
remember S in SOLID? is it responsibility of **parse** method to **save** the file?